code/modules/mob/living/silicon/robot/robot.dm

```diff
@@ -285,12 +285,12 @@
 	if(!ionpulse_on)
 		return
 
-	if(cell.charge <= 50)
+	if(cell.charge <= 10)
 		toggle_ionpulse()
 		return
 
-	cell.charge -= 50 // 500 steps on a default cell.
-	return 1
+	cell.charge -= 10
```

Comment:
Maybe this can be a variable somewhere in the ionpulse upgrade?

Reply:
what

Reply:
I think he means making a variable in the upgrade itself such as cell_drain and having this code reference that variable when draining so it can be modified ingame more easily or having a subtype made with a different drain